corgi/tasks/prod_defs.py

```diff
@@ -215,19 +215,46 @@ def parse_product_version(
         _match_and_save_stream_cpes(product_version)
 
 
-def _parse_variants_from_brew_tags(brew_tags) -> list[str]:
+def _parse_variants_from_brew_tags(brew_tags: list[str]) -> dict[str, str]:
     """We don't link core ProductVariant models to streams via brew_tags because it results in too
     many builds being linked to the stream. These variants should include all possible builds that
     could match the stream. We use them for finding which errata match this stream, so it's OK to
     find too many errata"""
-    variant_names: set[str] = set()
+    distinct_variants: dict[str, str] = {}
+    # TODO add CollectorErrataProductVariant.repos to ProductVariant
     for brew_tag in brew_tags:
         trimmed_brew_tag, sans_container_released = brew_tag_to_et_prefixes(brew_tag)
-        variants = CollectorErrataProductVersion.objects.filter(
+        variant_data = CollectorErrataProductVersion.objects.filter(
             brew_tags__overlap=[trimmed_brew_tag, sans_container_released]
-        ).values_list("variants__name", flat=True)
-        variant_names.update(variants)
-    return list(variant_names)
+        ).values("variants__name", "variants__cpe")
+        for variant in variant_data:
+            distinct_variants[variant["variants__name"]] = variant["variants__cpe"]
```

Comment:
Should we exclude variants without any CPE?

You can also use .values_list() here, then just unpack the tuple. We only care about the name / CPE values and not the keys themselves, so there's no need to ask for a dict:
```suggestion
        ).values_list("variants__name", "variants__cpe")
        for variant_name, variant_cpe in variant_data:
            if not variant_cpe:
                continue
            distinct_variants[variant_name] = variant_cpe
```

Reply:
ok